src/components/DisplayArticlesData.jsx

```diff
@@ -1,60 +1,28 @@
-import React, { useState, useEffect } from 'react';
-import { getArticleData } from '../modules/articlesData';
-import { Card } from 'semantic-ui-react';
+import React, { useEffect } from "react";
+import { getArticleData } from "../modules/articlesData";
+import ArticleCard from "./ArticleCard";
+import { useSelector, useDispatch } from "react-redux";
+import { Card } from "semantic-ui-react";
 
 const DisplayArticlesData = () => {
-  const [articlesData, setArticlesData] = useState([]);
-  const getArticles = async () => {
-    let result = await getArticleData();
-    setArticlesData(result.data.articles);
-  };
+  const dispatch = useDispatch();
+  const { mainArticles } = useSelector((state) => state);
 
   useEffect(() => {
-    getArticles();
-  }, []);
+    getArticleData.index(dispatch);
+  }, [dispatch]);
 
-  let articleIndex = articlesData.map((article) => {
-    return (
-      <Card key={article.id} data-cy={`article-${article.id}`}>
-        <Card.Content>
-          <Card.Header
-            data-cy="title"
-            style={{
-              fontWeight: 'bold',
-              fontSize: 30,
-              color: 'black',
-              textAlign: 'center',
-            }}
-          >
-            {article.title}
-          </Card.Header>
-          <Card.Meta
-            data-cy="sub-title"
-            style={{
-              fontWeight: 'bold',
-              fontSize: 20,
-              color: 'black',
-              textAlign: 'center',
-            }}
-          >
-            {article.sub_title}
-          </Card.Meta>
-          <Card.Description
-            data-cy="content"
-            style={{ fontWeight: 'small', fontSize: 15, color: 'black' }}
-          >
-            {article.content}
-          </Card.Description>
-          <Card.Meta data-cy="author-email">{article.author}</Card.Meta>
-          <Card.Meta data-cy="created-at">{`Created at: ${article.created_at}`}</Card.Meta>
-          <Card.Meta data-cy="updated-at">{`Updated at: ${article.updated_at}`}</Card.Meta>
-        </Card.Content>
-      </Card>
-    );
-  });
+  let articleIndex;
+  articleIndex = (
+    <Card.Group itemsPerRow={3}>
+      {mainArticles.map((article) => {
+        return <ArticleCard article={{ ...article }} />;
+      })}
+    </Card.Group>
+  );
 
   return (
-    <>{articlesData.length > 0 && <div data-cy="index">{articleIndex}</div>}</>
+    <>{mainArticles.length > 0 && <ul data-cy="index">{articleIndex}</ul>}</>
```

Comment:
why are you using an `ul` when the `articleIndex` does not contain list items but cards?

Reply:
changed it to a `div`